src/models/relations.ts

```diff
@@ -251,7 +251,12 @@ export class Relations extends TypedEventEmitter<RelationsEvent, EventHandlerMap
      * @param {MatrixEvent} redactedEvent
      * The original relation event that is about to be redacted.
      */
-    private onBeforeRedaction = async (redactedEvent: MatrixEvent): Promise<void> => {
+    private onBeforeRedaction = (redactedEvent: MatrixEvent): void => {
+        // TODO: why are we not handling errors?
+        void this.onBeforeRedactionPromise(redactedEvent);
+    };
+
+    private onBeforeRedactionPromise = async (redactedEvent: MatrixEvent): Promise<void> => {
```

Comment:
I think naming a function `...Promise` is misleading

Reply:
I've committed an alternative way to do this.